### ISSUE TITLE
Closing the main window does not terminate the application on Linux

### ISSUE THREAD

**Issue by netvl** [standardnotes/desktop#14]
I've downloaded and run the appimage file:
```
% chmod +x Standard-Notes-Linux-0.1.1-x86_64.AppImage
% ./Standard-Notes-Linux-0.1.1-x86_64.AppImage
```
The program started normally (except a message about a coredump of zenity, but as far as I can see in the appimage launch script, it is nothing of importance), however, closing the main window did not stop the program: there is still quite a lot spawned processes. I have to press Ctrl+C in the terminal to kill them. This is not acceptable for me because I don't want to launch the app through the console, I want to do it from my window manager's launch prompt.

Also, as a side question, are there any plans to provide an alternative packaging variants for linux? Or at least an ability to assemble something else than appimage? I would prefer Standard Notes to be installed to my system directly and to be accessible in PATH, like all other programs. There seem to be no easy way to extract an appimage image and produce a package out of it, since it uses a very weird file layout. My system is Archlinux, by the way.

**Comment by mobitar** [comment]
On Debian platforms (like Mac), close the window doesn't kill it (as that's default behavior). Not sure if Archlinux is a descendant of Debian. Can you point me to some resources as to best way to distribute linux apps? There doesn't seem to be consensus on the matter.

**Comment by netvl** [comment]
> closing the window doesn't kill it (as that's default behavior)

Well, yes, that's the problem. How do I quit the application otherwise? I didn't see any "Quit" operations in menus or something like it.

No, Archlinux is not a Debian descendant. And unfortunately, no, I'm not aware of general packaging guidelines for Linux. I believe the most common way to distribute an application is to create deb and rpm packages, and other distributions may then use them to prepare their own packages. Lots of packages in AUR for electron-based apps do that.

Or, alternatively, there could be a tarball with all the files put into correct places (the main binary in /usr/bin, icons and other resources in /usr/share, etc.); this could be made into any other package by anyone who would want to maintain the package in a distribution.

Finally, for lots of applications which use build systems like autotools or CMake or other similar things, there may be no need to do special packaging at all, since these tools usually already provide means to form the necessary file tree. To build a package for such programs, it is sufficient to download the source tarball/clone the repository and run the build system in some way specific to the distribution packaging tools (which, naturally, is done by the package maintainers). I don't know if this is possible or feasible for electron-based applications.

It probably makes sense to look how other applications like Atom, VSCode, N1 or Simplenote (some electron apps I know, there has got to be lots more) prepare their Linux packages in various distributions.

**Comment by mobitar** [comment]
Got it, will make note of both these points for next update. Thanks for the tip.

**Comment by netvl** [comment]
That being said, I don't think that macOS (which has Darwin kernel) is a Debian descendant (which is based on Linux) :)

As for quitting behavior, I believe most applications in Linux are similar to Windows - when the last window of an application is closed, it is expected that the entire application is terminated, because neither in Linux desktop environments (two common exceptions are probably Unity and Gnome 3, but even though they have large user base, they still are relatively recent, and even in them applications tend not to persist past their last window) nor in Windows (at least, until 7, when the task bar was changed to be more dock-like) there were any kind of "dock" like in macOS, so there is no way for an application to be accessible without opened windows. There is usually some kind of system tray which contains icons for some resident applications, but its user experience is completely different from the one provided by the dock in macOS, and applications are usually not expected to put a tray icon unless they are supposed to be residential (like messengers or antiviruses). It *is* probably a good idea to support tray icons for applications like Standard Notes (for example, Evernote does have a tray icon in Windows and even in Mac, if I recall correctly), though, but this is certainly not an absolutely necessary feature.

**Comment by mobitar** [comment]
Sorry, totally meant to say Darwin (not Debian). But now that I double check the code, I do actually see that I'm not performing a check for Darwin and quitting is being prevented. Thanks for bringing this up. Will get a fix out for this in the next release.

**Comment by andykais** [comment]
a simple solution would be to put `process.exit(0)` in the [win.on('close') event](https://github.com/standardnotes/desktop/blob/master/app/index.js#L48).

**Comment by mobitar** [comment]
Fixed via a5e9a0632dc8ae300dcad65a1a909dd4e5bfb5a6
